### PR TITLE
Fix Project Create/Update/Delete Modals

### PR DESCRIPTION
**Pull request by andrewballantyne**
Closes: #1239
Closes: #1240

## Description

Cleanup the Project modal:
* Create Project is waiting on the project being fetched -- fetches all projects until the project is available
* Update Project shows errors
* Both Project modals refreshes inside the modal
* Delete modals now wait on closure before clearing the content

## How Has This Been Tested?

Manually. Spent most of the time testing Projects List View and creating, deleting, updating modals.

## Test Impact

None

## Request review criteria:

- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

File: frontend/src/pages/projects/screens/projects/ProjectListView.tsx [hunk 1]
```diff
@@ -109,26 +108,27 @@ const ProjectListView: React.FC<ProjectListViewProps> = ({ allowCreate }) => {
       />
       <ManageProjectModal
         open={!!editData}
-        onClose={() => {
+        onClose={(newProjectName) => {
```
Comment (Gkrumbach07) [inline frontend/src/pages/projects/screens/projects/ProjectListView.tsx:111]: this doesnt get called for some reason and the modal doesnt get closed on cancel of the modal

**Review by Gkrumbach07 — COMMENTED** [review]
code looks good. except for the modal close issue

File: frontend/src/pages/projects/screens/projects/ProjectListView.tsx (hunk 1 above)
Comment (lucferbux) [reply frontend/src/pages/projects/screens/projects/ProjectListView.tsx:111]: No, cause the modal is opened in `NewProject` when opened in the Project pages.

**Review by lucferbux — CHANGES_REQUESTED** [review]

File: frontend/src/pages/projects/screens/projects/NewProjectButton.tsx [hunk 2]
```diff
@@ -18,19 +16,16 @@ const NewProjectButton: React.FC<NewProjectButtonProps> = ({ closeOnCreate, onPr
...
       <ManageProjectModal
-        onProjectCreated={(projectName) => {
-          refresh().then(() => {
-            if (closeOnCreate) {
-              setOpen(false);
-            }
+        open={open}
+        onClose={(newProjectName) => {
+          if (closeOnCreate) {
+            setOpen(false);
```
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/src/pages/projects/screens/projects/NewProjectButton.tsx]: I am aware, thanks... it wasn't a confusion on the bug it was a lack of time to fix it haha.

**Comment by Gkrumbach07** [comment]
looks good now /lgtm

**Pull request merged**